File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
    var number: Int

    init(_ number: Int) {
        self.number = number
    }

    var squareOfSum: Int {
        return (1 ... number).reduce(0, +) * (1 ... number).reduce(0, +)
    }

    var sumOfSquares: Int {
        return (1 ... number).map { $0 * $0 }.reduce(0, +)
    }

    var differenceOfSquares: Int {
        return squareOfSum - sumOfSquares
    }
}

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var total: usize = 0;
    for (0..number + 1) |n| {
        total += n;
    }

    return total * total;
}

pub fn sumOfSquares(number: usize) usize {
    var total: usize = 0;
    for (0..number + 1) |n| {
        total += n * n;
    }

    return total;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
